Add TransportAdapter parameter to PyCrest

Kyria opened the pull request:
Add the transport adapter to the EVE class as a constructor parameter, this way, as we discussed in #42, the user will be able to define it the way he need it, or simply use the requests default transport adapter `HTTPAdapter`

Kyria committed:
Add new parameter to EVE class to specify the transport adapter we want to use.
This will allow the users to manually specify the transport adapter, with all the options they need

Kyria committed:
Add new parameter to EVE class to specify the transport adapter we want to use.
This will allow the users to manually specify the transport adapter, with all the options they need

Kyria committed:
No cache option for PyCrest __init__ and get() calls (#47)

* Add DummyCache to fake a cache, not to break existing code but still allow a "no cache" configuration.
Also set DummyCache as default cache (cache=None).
Change existing tests to match this new behavior, and add test for DummyCache

* set DictCache as default, so we manually have to set it to none for no cache

* Add caching argument to __call__ for ApiConnection and ApiObject

* Remove prints...

* Add caching parameters to the doc

* Fix deprecated aliases for assertEqual (was using assertEquals)

* Remove another forgotten print in tests

hkraal commented:
@Kyria
I'm good to merge after the conflict is solved and one small addition:
- Can you add a sanity check that the `transport_adapter=` parameter contains an subclassed instance of `requests.adapters.HTTPAdapter` (if I'm not mistaken) to prevent things like `transport_adapter=''` for example?

Kyria committed:
Resolve conflicts from the PR with the cache modification.

Conflicts:
	pycrest/eve.py

Kyria committed:
Add check to verify the transport adapter is a instance from HTTPAdapter or a subclass
Also add tests to check this.

Kyria commented:
@hkraal 
I've added a check for HTTPAdapter (and subsets) instance : `isinstance(transport_adapter, HTTPAdapter):`
I was tempted to do the check on BaseAdapter (as HTTPAdapter inherits from it), but as we want to do some HTTP/S stuffs, and as HTTPAdapter is the default requests adapter, I made the choice to do this.

I also added some test to test these "fail" cases (wrong adapter / empty string adapter). 

Nothing about the content itself, but the PR is a bit weird in the "file changed" tab after I updated the branch with the cache PR you merged... It shows cache commits, but I don't see them when comparing branches https://github.com/pycrest/PyCrest/compare/master...Kyria:transport-adaptor-param so it looks good :)

jonobrien commented on pycrest/eve.py:
```diff
@@ -159,7 +159,7 @@ def __init__(
             user_agent = "PyCrest/{0} +https://github.com/pycrest/PyCrest"\
                 .format(version)
-        if transport_adapter is not None:
+        if transport_adapter is not None and isinstance(transport_adapter, HTTPAdapter):
```
Could this also be `if transport_adapter and isinstance...`  since you would need something to not be None when isinstance checks anyway?

hkraal replied:
Is could even be

``` python
if isinstance(transport_adapter, HTTPAdapter):
    pass
```

as `None` will be an instance of NoneType

@Kyria I will review the PR after Monday as I'm currently away with family

Kyria replied:
Ahah, right, I don't even know why I left that `if transport_adapter is not None` when I added the check on `isinstance` as checking isinstance is enough...

Kyria committed:
remove None check for transport_adapter as it's not required with isinstance

hkraal commented:
Merged! :)
